app/models/transaction.rb

```diff
@@ -0,0 +1,24 @@
+class Transaction < ActiveRecord::Base
+  before_save :set_version, :create_version, if: :changed?
+
+  validates :uid, :booked_at, :booking_at, :booking_status, :delivery_partner,
+            presence: true
+  validates :uid, uniqueness: true
+
+  scope :transactions, ->(end_point) { where(['booking_at <= ?', end_point]) }
+  scope :bookings, ->(end_point) { where(['booked_at <= ?', end_point]).where(cancelled: false) }
+  scope :completed_transactions, ->(end_point) { transactions(end_point).where(booking_status: 'Complete') }
+  scope :new_today, -> { where(updated_at: Time.zone.today..Time.zone.now, version: 1) }
+
+  def set_version
```

Comment:
May make sense to default the `version` to `0` and always increment on `before_save`?